File: frontend/src/api/onboarding.ts
import apiClient from './client';
import { RoutineBlock, ParentGoalData } from '../types';

export interface SurveyPayload {
  child_name?: string;
  child_age?: number;
  diagnosis_year?: number;
  communication_level?: string;
  challenges?: string[];
  strengths?: string[];
  strengths_freetext?: string;
  wake_time?: string;
  school_start?: string;
  school_end?: string;
  bedtime?: string;
  screen_time?: string;
  parent_goals?: string[];
  support_system?: string[];
  current_step?: number;
}

export const onboardingApi = {
  setLanguage: async (language: 'en' | 'tl') => {
    const { data } = await apiClient.post('/onboarding/language', { language });
    return data;
  },

  saveSurvey: async (payload: SurveyPayload) => {
    const { data } = await apiClient.post('/onboarding/survey', payload);
    return data;
  },

  getSurvey: async (): Promise<SurveyPayload> => {
    const { data } = await apiClient.get('/onboarding/survey');
    return data;
  },

  complete: async (): Promise<{
    routine: RoutineBlock[];
    goals: ParentGoalData[];
    child_id?: string;
  }> => {
    const { data } = await apiClient.post('/onboarding/complete');
    return data;
  },

  getRoutine: async (childId?: string): Promise<RoutineBlock[]> => {
    const { data } = await apiClient.get('/onboarding/routine', {
      params: childId ? { child_id: childId } : undefined,
    });
    return data;
  },

  getGoals: async (childId?: string): Promise<ParentGoalData[]> => {
    const { data } = await apiClient.get('/onboarding/goals', {
      params: childId ? { child_id: childId } : undefined,
    });
    return data;
  },
};

File: backend/app/schemas/onboarding.py
from typing import Optional, List
from pydantic import BaseModel


class LanguageRequest(BaseModel):
    language: str   # "en" | "tl"


class SurveyPayload(BaseModel):
    child_name: Optional[str] = None
    child_age: Optional[int] = None
    diagnosis_year: Optional[int] = None
    communication_level: Optional[str] = None
    challenges: List[str] = []
    strengths: List[str] = []
    strengths_freetext: Optional[str] = None
    wake_time: Optional[str] = None
    school_start: Optional[str] = None
    school_end: Optional[str] = None
    bedtime: Optional[str] = None
    screen_time: Optional[str] = None
    parent_goals: List[str] = []
    support_system: List[str] = []
    current_step: int = 0


class RoutineBlockOut(BaseModel):
    time_range: str
    activity_key: str
    duration_minutes: int
    category: str
    tip_key: str
    why_key: str
    icon_key: str


class ParentGoalOut(BaseModel):
    goal_key: str
    order: int
    frequency: str
    target_days: int


class OnboardingCompleteResponse(BaseModel):
    routine: List[RoutineBlockOut]
    goals: List[ParentGoalOut]
    child_id: Optional[str] = None
